src/core/exec-invoke.c

```diff
@@ -1098,6 +1098,26 @@ static int null_conv(
         return PAM_CONV_ERR;
 }
 
+static int pam_close_session_and_delete_credentials(pam_handle_t *handle, int flags) {
+        int r;
+
+        assert(handle);
+
+        r = pam_close_session(handle, flags);
+        if (r != PAM_SUCCESS) {
+                log_debug("pam_close_session() failed: %s", pam_strerror(handle, r));
+                return r;
```

Comment:
hmm, I think we probably should still do the pam_setcred call below if pam_close_session fails, no? i.e. a RET_GATHER() style situation, except with pam errors...